pkg/genericapiserver/api/handlers/responsewriters/status.go

```diff
@@ -49,20 +48,14 @@ func apiStatus(err error) *metav1.Status {
 		//TODO: check for invalid responses
 		return &status
 	default:
-		status := http.StatusInternalServerError
-		switch {
-		//TODO: replace me with NewConflictErr
-		case storage.IsTestFailed(err):
```

Comment:
Why do(did) we need this? We have proper `StorageError` translation in `InterpretUpdateError` and friends.